common/src/main/java/com/google/auto/common/BasicAnnotationProcessor.java

```diff
@@ -505,74 +536,391 @@ private ImmutableSetMultimap<Class<? extends Annotation>, Element> toClassKeyedM
+    }
+
+    @Override
+    protected boolean isAcceptableElementKind(Element element) {
+      return element.getKind() == PACKAGE;
+    }
+
+    @Override
+    Optional<PackageElement> getElement(Elements elementUtils) {
+      return Optional.ofNullable(elementUtils.getPackageElement(toString));
+    }
+  }
+
+  private static final class TypeElementBlueprint extends PackageOrTypeElementBlueprint {
+    private TypeElementBlueprint(Element element) {
+      super(element);
+    }
+
+    @Override
+    protected boolean isAcceptableElementKind(Element element) {
+      return isType(element);
+    }
+
+    @Override
+    Optional<TypeElement> getElement(Elements elementUtils) {
+      return Optional.ofNullable(elementUtils.getTypeElement(toString));
+    }
+  }
+
+  private static final class TypeParameterElementBlueprint extends ElementBlueprint {
+    private final TypeElementBlueprint enclosingTypeElementBlueprint;
+
+    private TypeParameterElementBlueprint(Element element) {
+      super(element);
+      this.enclosingTypeElementBlueprint = new TypeElementBlueprint(element.getEnclosingElement());
+    }
+
+    @Override
+    protected boolean isAcceptableElementKind(Element element) {
+      return element.getKind() == TYPE_PARAMETER;
+    }
+
+    @Override
+    Optional<? extends TypeParameterElement> getElement(Elements elementUtils) {
+      return enclosingTypeElementBlueprint
+          .getElement(elementUtils)
+          .map(
+              enclosingTypeElementBlueprint ->
+                  enclosingTypeElementBlueprint.getTypeParameters().stream()
+                      .filter(typeParamElement -> toString.equals(typeParamElement.toString()))
+                      .collect(MoreCollectors.onlyElement()));
+    }
+
+    @Override
+    public boolean equals(@Nullable Object object) {
+      if (this == object) {
+        return true;
+      } else if (!(object instanceof TypeParameterElementBlueprint)) {
+        return false;
+      }
+
+      TypeParameterElementBlueprint that = (TypeParameterElementBlueprint) object;
+      return this.toString.equals(that.toString)
+          && this.enclosingTypeElementBlueprint.equals(that.enclosingTypeElementBlueprint);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(toString, enclosingTypeElementBlueprint);
+    }
+  }
+
+  /** Represents FIELD, ENUM_CONSTANT, and RECORD_COMPONENT */
+  private abstract static class FieldOrRecordComponentElementBlueprint extends ElementBlueprint {
+    private final TypeElementBlueprint enclosingTypeElementBlueprint;
+    private final ElementKind elementKind;
+
+    private FieldOrRecordComponentElementBlueprint(Element element) {
+      super(element); // toString is its simple name.
+      this.enclosingTypeElementBlueprint = new TypeElementBlueprint(getEnclosingType(element));
+      this.elementKind = element.getKind();
+    }
+
+    @Override
+    Optional<VariableElement> getElement(Elements elementUtils) {
```

Comment:
This isn't correct for a `RecordComponentElement`, which isn't a `VariableElement`.

Reply:
Well spotted, thank you. I must admit at the time I was writing this in 2022, I just started working with Java 11 and did not know about records.